src/store/indexedDB.js

```diff
@@ -0,0 +1,69 @@
+import encryptionService from "@src/services/encryptionService";
+import Dexie from "dexie";
+
+class IndexedDB {
+  constructor() {
+    const db = new Dexie("samadb");
+    db.version(1).stores({
+      messages: "++_id, cid, created_at",
+    });
+    this.db = db;
+  }
+
+  markMessagesAsRead(mids) {
+    this.db.messages.bulkUpdate(
+      mids.map((id) => ({ key: id, changes: { status: "read" } }))
+    );
+  }
+
+  async addMessage(message) {
+    const encryptedBody = encryptionService.encrypteDataForLocalStore(
```

Comment:
In a case we are in a regular chat (not encrypted), how this method will work? 
For regular chats we just need to store a message, w/o encrypting it